src/client/Microsoft.Identity.Client/Cache/CacheFallbackOperations.cs

```diff
@@ -318,6 +318,13 @@ internal static class CacheFallbackOperations
             // if we have more than 1 RT per env, there is smth wrong with the ADAL cache
             if (rtGroupsByEnv.Any(g => g.Count() > 1))
             {
+                //Due to the fact that there is a problem with the ADAL cache that causes this exception, The ADAL cache will be removed when 
+                //this exception is triggered so that the can sign in interactivly and repopulate the cache.
+                IDictionary<AdalTokenCacheKey, AdalResultWrapper> adalCache =
+                    AdalCacheOperations.Deserialize(logger, legacyCachePersistence.LoadCache());
+
+                adalCache.Clear();
+
                 throw new MsalClientException(
```

Comment:
I know we discussed throwing a UI required exception but this is already throwing this so it would result in a breaking change right? @bgavrilMS

Reply:
It is theoretically a breaking change indeed. But in practice, most apps already implement the pattern: 

try 
  - AquireTokenSiletn
catch UiRequiredEx
  - AcquireTokenInteractive

So I think throwing UIRequiredEx is still the right thing to do.

Reply:
Most ADAL operations are implemented inside a `try / catch all `, but this one is an exception. So here's my suggestion

1. Instead of throwing, just log the error
2. return null. This will cause AcquireTokenSilent to not find any RTs, which makes it throw an MsalUiRequiredException, which is smth apps know how to handle

Reply:
And also clear the ADAL cache, as it's broken.